### ISSUE TITLE
reflog doesn't include "(merge)" for merge commits

### ISSUE THREAD

**Issue by rcjsuen** [libgit2/libgit2#4094]
When you use the command line, merge commits will have `commit (merge):` recorded in the reflog as a reminder that a merge commit was created. libgit2 doesn't do this though. It only handles the simple `commit:` case and the first `commit (initial):` case.

### Reproduction steps
1. Create a merge commit using libgit2's APIs.
2. Check the output of `git reflog`.
### Expected behavior
`cf97c9d HEAD@{0}: commit (merge): Merge branch 'other'`
### Actual behavior
`d50f21e HEAD@{0}: commit: Merge branch 'other'`
### Version of libgit2 (release number or SHA1)
Looks like the code is still the same on 8d3b39a6ad325fe1d08e17281803e26359756fc6.
https://github.com/libgit2/libgit2/blob/8d3b39a6ad325fe1d08e17281803e26359756fc6/src/refs.c#L1148-L1153
### Operating system(s) tested
Windows 10

**Comment by ethomson** [comment]
Thanks for the report @rcjsuen !

This seems like a straightforward fix, I'm going to add the "Easy Fix" and "Up for Grabs" tags in case someone wants to submit a pull request!

**Comment by ratlifri** [comment]
I was looking here at #4094 this week and was coming toward a similar conclusion but above fix is better than mine.  I am was happy to learn I was in the right spot and looking to change the same field.  I added above fix into v0.25.1 (2fcb870) and all tests passed on Windows 7.

I looked at adding a new test and found that reflog is not tested very much, at least in this fashion, for anything besides "commit (initial)" which failed only four times after breaking on purpose, 3 in rebase::merge and 1 in commit::write.  What I think I learned about a merge commit from testing point of view is it crosses unit test boundaries since merge puts repo in a merge state and until the commit completes.
